src/App.tsx

```diff
@@ -0,0 +1,51 @@
+// IMPORTS -----------------------------------
+/* libraries */
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import styled from 'styled-components';
+/* components */
+import Filters from './Filters';
+import Products from './Products';
+/* custom hooks */ 
+import useProducts from './customHooks/useProducts';
+/* types */
+import { CategoryName } from './types';
+
+// COMPONENT -----------------------------------
+const App: React.FC = () => {
+  // set funtional component state
+  const { products, addProducts } = useProducts([]);
+  const [ filter, setFilter ] = useState<any>('All Products');
+
+  function handleSelect(newFilter: CategoryName) {
+    setFilter(newFilter);
+  }
+
+  useEffect(() => {
+    // asynchronously fetch/set products
+    const fetchProducts = async () => {
+      const response = await axios.get("http://localhost:9001/products");
+      addProducts(response.data);
+      console.log("Products added!");
+    }
```

Comment:
This function could be declared outside of the scope of useEffect. In this implementation, fetchProducts will be reinstantiated every useEffect call, which is not the most efficient. I would put this between handleSelect and UseEffect.like this:

```
function handleSelect(newFilter: CategoryName) {
    setFilter(newFilter);
}

const fetchProducts = async () => {
...
}

useEffect(fetchProducts, [])
```